x/auth/ante.go

```diff
@@ -95,8 +96,15 @@ func NewAnteHandler(am AccountMapper, fck FeeCollectionKeeper) sdk.AnteHandler {
 			}
 
 			// first sig pays the fees
-			// TODO: Add min fees
 			// Can this function be moved outside of the loop?
+			if ctx.IsCheckTx() && !simulate {
+				// required fees must be greater than the minimum set by the validator
+				if ctx.MinimumFees().Minus(fee.Amount).IsNotNegative() {
+					fee = NewStdFee(fee.Gas, ctx.MinimumFees()...)
+				}
+				// validators reject any tx from the mempool with less than the required fee per gas * gas factor
+				fee = NewStdFee(fee.Gas, calculateFeesByGas(fee.Amount, fee.Gas)...)
```

Comment:
A `StdFee` is only composed of an `Amount` and `Gas`, why not just pass a single `fee` type into `calculateFeesByGas`?

Reply:
I thought that `calculateFeesByGas` didn't necessarily need to build a new `StdFee`

Reply:
You're right, it doesn't, but the arguments to `calculateFeesByGas` are all contained in your `fee` variable. Im suggesting, why not just pass the `fee` variable...in other words, cleaning up the function IMHO.

Reply:
Refactored accordingly, thanks